File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s=s.toLowerCase();
        String ss= s.replaceAll("[^a-zA-Z0-9]", "");
        Stack<String> stack=new Stack<>();
        // for(int i=0;i<s.length();i++){
        //     Stack.push(s.charAt(i));
        // }
        for(int i=0;i<ss.length()/2;i++){
            if(ss.charAt(i)!=ss.charAt(ss.length()-1-i)){
                return false;
            }
        }
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        new = ''.join(c.lower() for c in s if c.isalnum())
        L=0
        R=len(new)-1

        while L<R:
            if new[L]!= new[R]:
                return False
            L+=1
            R-=1
        return True